minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/RunMiNiFi.java

```diff
@@ -1080,15 +1111,31 @@ public void start() throws IOException, InterruptedException {
 
                     final File reloadFile = getReloadFile(defaultLogger);
                     if (reloadFile.exists()) {
-                        defaultLogger.info("Currently reloading configuration.  Will not restart MiNiFi.");
+                        defaultLogger.info("Currently reloading configuration. Will wait to restart MiNiFi.");
                         Thread.sleep(5000L);
                         continue;
                     }
 
                     final boolean previouslyStarted = getNifiStarted();
                     if (!previouslyStarted) {
-                        defaultLogger.info("MiNiFi never started. Will not restart MiNiFi");
-                        return;
+                        final File swapConfigFile = getSwapFile(defaultLogger);
+                        if(swapConfigFile.exists()){
+                            defaultLogger.info("Swap file exists, MiNiFi failed trying to change configuration. Reverting to old configuration.");
+
+                            final String confDir = getBootstrapProperties().getProperty(CONF_DIR_KEY);
+                            ((MiNiFiConfigurationChangeListener) changeListener).performTransformation(new FileInputStream(swapConfigFile), confDir);
```

Comment:
Would prefer to just declare the field of this class to just be a MiNiFiConfigurationChangeListener instead of having to do the cast.